src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -267,6 +328,56 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
     return response;
 }
 
+cJSON* wm_task_manager_build_response_result(cJSON *res, const char *module, const char *command, char *status, int create_time, int last_update_time, char *request_command) {
+    
+    if (module != NULL) {
+        cJSON_AddStringToObject(res, json_keys[WM_TASK_MODULE], module);
+    }
+    if (command != NULL) {
+        cJSON_AddStringToObject(res, json_keys[WM_TASK_COMMAND], command);
+    }
+    if (status != NULL) {
+
+        if (!strcmp(commands_list[WM_TASK_UPGRADE_RESULT], request_command)){
+            
+            if (!strcmp(task_statuses[WM_TASK_DONE], status)){
```

Comment:
Is it any way we can map the task_status with the upgrade_status to avoid all this conditionals?